src/Fixer/Operator/NoUselessConcatOperatorFixer.php

```diff
@@ -0,0 +1,306 @@
+use PhpCsFixer\FixerDefinition\FixerDefinition;
+use PhpCsFixer\FixerDefinition\FixerDefinitionInterface;
+use PhpCsFixer\Preg;
+use PhpCsFixer\Tokenizer\Token;
+use PhpCsFixer\Tokenizer\Tokens;
+
+final class NoUselessConcatOperatorFixer extends AbstractFixer implements ConfigurableFixerInterface
+{
+    private const STR_DOUBLE_QUOTE = 0;
+    private const STR_DOUBLE_QUOTE_VAR = 1;
+    private const STR_SINGLE_QUOTE = 2;
+
+    /**
+     * {@inheritdoc}
+     */
+    public function getDefinition(): FixerDefinitionInterface
+    {
+        return new FixerDefinition(
+            'There should not be useless concat operations.',
+            [
+                new CodeSample("<?php\n\$a = 'a'.'b';\n"),
+                new CodeSample("<?php\n\$a = 'a'.\"b\";\n", ['juggle_simple_strings' => true]),
+            ],
+        );
+    }
+
+    /**
+     * {@inheritdoc}
+     *
+     * Must run before DateTimeCreateFromFormatCallFixer, EregToPregFixer, PhpUnitDedicateAssertInternalTypeFixer, RegularCallableCallFixer, SetTypeToCastFixer.
+     * Must run after NoBinaryStringFixer, SingleQuoteFixer.
+     */
+    public function getPriority(): int
+    {
+        return 5;
+    }
+
+    /**
+     * {@inheritdoc}
+     */
+    public function isCandidate(Tokens $tokens): bool
+    {
+        return $tokens->isTokenKindFound('.') && $tokens->isAnyTokenKindsFound([T_CONSTANT_ENCAPSED_STRING, '"']);
+    }
+
+    /**
+     * {@inheritdoc}
+     */
+    protected function applyFix(\SplFileInfo $file, Tokens $tokens): void
+    {
+        for ($index = $tokens->count() - 1; $index > 0; --$index) {
+            if (!$tokens[$index]->equals('.')) {
+                continue;
+            }
+
+            $nextMeaningfulTokenIndex = $tokens->getNextMeaningfulToken($index);
+
+            if ($this->containsLinebreak($tokens, $index, $nextMeaningfulTokenIndex)) {
+                continue;
+            }
+
+            $secondOperand = $this->getConcatOperandType($tokens, $nextMeaningfulTokenIndex, 1);
+
+            if (null === $secondOperand) {
+                continue;
+            }
+
+            $prevMeaningfulTokenIndex = $tokens->getPrevMeaningfulToken($index);
+
+            if ($this->containsLinebreak($tokens, $prevMeaningfulTokenIndex, $index)) {
+                continue;
+            }
+
+            $firstOperand = $this->getConcatOperandType($tokens, $prevMeaningfulTokenIndex, -1);
+
+            if (null === $firstOperand) {
+                continue;
+            }
+
+            $this->fixConcatOperation($tokens, $firstOperand, $index, $secondOperand);
+        }
+    }
+
+    /**
+     * {@inheritdoc}
+     */
+    protected function createConfigurationDefinition(): FixerConfigurationResolverInterface
+    {
+        return new FixerConfigurationResolver([
+            (new FixerOptionBuilder('juggle_simple_strings', 'Allow for simple string quote juggling if it results in more concat-operations merges.'))
```

Comment:
What are the use-cases of this option? Can't we drop it and always enable this behavior?

Reply:
I would really like the fixer to do the juggling in my projects, however most project will likely not adopt it.
For example in this project:

```diff
   3) PHP-CS-Fixer/tests/Smoke/CiIntegrationTest.php (no_useless_concat_operator)
      ---------- begin diff ----------
--- /home/possum/work/PHP-CS-Fixer/tests/Smoke/CiIntegrationTest.php
+++ /home/possum/work/PHP-CS-Fixer/tests/Smoke/CiIntegrationTest.php
@@ -45,13 +45,13 @@
         try {
             CommandExecutor::create('composer --version', __DIR__)->getResult();
         } catch (\RuntimeException $e) {
-            static::markTestSkippedOrFail('Missing `composer` env script. Details:'."\n".$e->getMessage());
+            static::markTestSkippedOrFail("Missing `composer` env script. Details:\n".$e->getMessage());
         }
 
         try {
             CommandExecutor::create('composer check', __DIR__.'/../..')->getResult();
         } catch (\RuntimeException $e) {
-            static::markTestSkippedOrFail('Composer check failed. Details:'."\n".$e->getMessage());
+            static::markTestSkippedOrFail("Composer check failed. Details:\n".$e->getMessage());
         }
```

(there are at least 9 files like this in this project alone)